benchmark/solver/solver.cpp

```diff
@@ -469,12 +536,18 @@ int main(int argc, char *argv[])
             std::clog << "Matrix is of size (" << system_matrix->get_size()[0]
                       << ", " << system_matrix->get_size()[1] << ")"
                       << std::endl;
+            auto precond_solver_name = begin(precond_solvers);
             for (const auto &solver_name : solvers) {
-                solve_system(solver_name.c_str(), exec, system_matrix, lend(b),
-                             lend(x), test_case, allocator, rhs_engine);
-                std::clog << "Current state:" << std::endl
-                          << test_cases << std::endl;
-                backup_results(test_cases);
+                for (const auto &precond_name : preconds) {
+                    std::clog << "\tRunning solver: " << *precond_solver_name
+                              << std::endl;
+                    solve_system(solver_name, precond_name,
+                                 precond_solver_name->c_str(), exec,
+                                 system_matrix, lend(b), lend(x), test_case,
+                                 allocator, rhs_engine);
+                    backup_results(test_cases);
+                    ++precond_solver_name;
+                }
```

Comment:
Just so that I understand correctly: in terms of json data, for each matrix there will be different entries named `solver` or `solver-preconditionner` each with their respective data?

Reply:
Yes.